nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/CSVExportAction.java

```diff
@@ -0,0 +1,309 @@
+            String commandId = commandIdFrom(record);
+            Blob blob = getBlob(dataFrom(record));
+            try {
+                blob = BlobUtils.zip(blob, null);
+            } catch (IOException e) {
+                getLog().error(e, e);
+            }
+            storeBlob(blob, commandId);
+            context.produceRecord("o1", BulkRecords.of(commandId, countFrom(record), getKey(commandId)));
+            context.askForCheckpoint();
+        }
+
+    }
+
+    public static class ExposeBlobInDownload extends AbstractTransientBlobComputation {
+
+        public static final String NAME = "moveBlobToDownload";
+
+        public ExposeBlobInDownload() {
+            super(NAME);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String documentIdsStreamName, Record record) {
+            String commandId = commandIdFrom(record);
+            Blob blob = getBlob(dataFrom(record));
+            if (blob != null) {
+                TransientStore download = Framework.getService(TransientStoreService.class)
+                                                   .getStore(DownloadService.TRANSIENT_STORE_STORE_NAME);
+                download.putBlobs(commandId, Collections.singletonList(blob));
+                download.setCompleted(commandId, true);
+            }
+            KeyValueStore kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(BULK_KV_STORE_NAME);
+            BulkStatus status = BulkCodecs.getBulkStatusCodec().decode(kvStore.get(commandId + STATUS));
+            BulkCounter counter = new BulkCounter(commandId, status.getCount());
+            context.produceRecord("o1", commandId, BulkCodecs.getBulkCounterCodec().encode(counter));
+            context.askForCheckpoint();
+        }
+
+    }
+
+    protected abstract static class AbstractTransientBlobComputation extends AbstractComputation {
+
+        public AbstractTransientBlobComputation(String name) {
+            super(name, 1, 1);
+        }
+
+        String getKey(String commandId) {
+            return metadata.name() + commandId;
+        }
+
+        protected Blob getBlob(String key) {
+            TransientStore store = Framework.getService(TransientStoreService.class).getStore(ACTION_NAME);
+            List<Blob> blobs = store.getBlobs(key);
+            Blob blob = blobs == null || blobs.isEmpty() ? null : blobs.get(0);
+            if (blob == null) {
+                getLog().error("Could not retrienve blob for key " + key);
+            }
+            return blob;
+        }
+
+        protected void storeBlob(Blob blob, String commandId) {
+            TransientStore store = Framework.getService(TransientStoreService.class).getStore(ACTION_NAME);
+            store.putBlobs(getKey(commandId), Collections.singletonList(blob));
+            store.setCompleted(getKey(commandId), true);
```

Comment:
We shouldn't always give `true` to completed, for instance in our context, blob stored before zip is not completed.

Reply:
they are stored with different keys, each of them is completed